crates/autopilot/src/shadow.rs

```diff
@@ -21,11 +21,20 @@ use {
     number::nonzero::U256 as NonZeroU256,
     primitive_types::{H160, U256},
     rand::seq::SliceRandom,
-    shared::token_list::AutoUpdatingTokenList,
+    shared::{metrics::LivenessChecking, token_list::AutoUpdatingTokenList},
     std::{cmp, time::Duration},
     tracing::Instrument,
 };
 
+pub struct Liveness;
+#[async_trait::async_trait]
+impl LivenessChecking for Liveness {
+    async fn is_alive(&self) -> bool {
+        // can we somehow check that we keep processing auctions?
+        true
```

Comment:
We could make the `Liveness` struct part of the shared server state and update a timestamp whenever some solver processed an auction.
Then `is_alive()` would check that this timestamp is not older than some healthy duration.